packages/core/src/util/txInspector.ts

```diff
@@ -24,35 +28,96 @@ export type StakeKeyRegistrationInspection = StakeAddressCertificate[];
 
 /**
- * Inspects a transaction for value (coins + assets) sent by the provided addresses.
+ * Inspects a transaction for values (coins + assets) in inputs
+ * containing any of the provided addresses.
  *
  * @param {Address[]} ownAddresses own wallet's addresses
- * @returns {Value} total value sent
+ * @param {TxAlonzo[]} historicalTxs wallet's historical transactions
+ * @returns {Value} total value in inputs
  */
-export const valueSentInspector: SendReceiveValueInspector = (ownAddresses: Address[]) => (tx: TxAlonzo) => {
-  if (!isOutgoing(tx, ownAddresses)) return { coins: 0n };
-  const sentOutputs = tx.body.outputs.filter((out) => !isAddressWithin(ownAddresses)(out));
-  return coalesceValueQuantities(sentOutputs.map((output) => output.value));
+export const totalInputValuesInspector: TotalInputValuesInspector = (ownAddresses, historicalTxs) => (tx) => {
+  const receivedInputs = tx.body.inputs.filter((input) => isAddressWithin(ownAddresses)(input));
+  const receivedInputsValues = receivedInputs
+    .map((input) => resolveInputValue(input, historicalTxs))
+    .filter((value): value is Value => !!value);
+
+  return coalesceValueQuantities(receivedInputsValues);
 };
 
 /**
- * Inspects a transaction for value (coins + assets) received by the provided addresses.
+ * Inspects a transaction for values (coins + assets) in outputs
+ * containing any of the provided addresses.
  *
  * @param {Address[]} ownAddresses own wallet's addresses
- * @returns {Value} total value received
+ * @returns {Value} total value in outputs
  */
-export const valueReceivedInspector: SendReceiveValueInspector = (ownAddresses: Address[]) => (tx: TxAlonzo) => {
-  if (isOutgoing(tx, ownAddresses)) return { coins: 0n };
+export const totalOutputValuesInspector: SendReceiveValueInspector = (ownAddresses) => (tx) => {
   const receivedOutputs = tx.body.outputs.filter((out) => isAddressWithin(ownAddresses)(out));
   return coalesceValueQuantities(receivedOutputs.map((output) => output.value));
 };
 
+/**
+ * Inspects a transaction to see if any of the addresses provided are included in a transaction input
+ * or if any of the rewards accounts or stake keys are included in a certificate
+ *
+ * @param {SentInspectorArgs} args array of addresses, stake key hashes and/or reward accounts
+ * @returns {boolean} true if any of the arguments provided were found in the transaction, false otherwise
+ */
+export const sentInspector: SentInspector =
+  ({ addresses, rewardAccounts, stakeKeyHashes }) =>
+  (tx: TxAlonzo) => {
+    const isAddressInInputs = addresses?.length && isSending(tx, addresses);
+    const isStakeKeyInCertificate =
+      stakeKeyHashes?.length &&
+      tx.body.certificates?.some((cert) => 'stakeKeyHash' in cert && stakeKeyHashes.includes(cert.stakeKeyHash));
+    const isRewardAccountInCertificate =
+      rewardAccounts?.length &&
+      tx.body.certificates?.some((cert) => 'rewardAccount' in cert && rewardAccounts.includes(cert.rewardAccount));
```

Comment:
I'm wondering if we should refactor `MirCertificate` to also use `stakeKeyHash` instead, like other certificates. The original refactor of `rewardAccount`->`stakeKeyHash` was needed to not require `networkId` in `coreToCsl`. MIR was left behind because it's conversion was not implemented. What do you think @rhyslbw ?

Also, we probably want to check `poolParameters.rewardAccount` - I assume you have to sign such certificate with your stake key too?

Reply:
So `stakeKeyHash` instead of `rewardAccount`? It's probably best to make it consistent and simplify this logic.